src/utilities/network/lookupApi.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import { SETTINGS_KEYS } from '../../settings';
+import { UIDatabase } from '../../database/index';
+
+const { SYNC_URL } = SETTINGS_KEYS;
+
+const RESOURCES = {
+  PATIENT: '/api/v4/patient',
+  PRESCRIBER: '/api/v4/prescriber',
+};
+
+const SEPARATORS = {
+  QUERY_STRING: '?',
+  QUERY_PARAMETERS: '&',
+  POLICY_NUMBER: '-',
+};
+
+const getQueryString = params => {
+  const {
+    firstName = '',
+    lastName = '',
+    registrationCode = '',
+    dateOfBirth = '',
+    policyNumberPerson = '',
+    policyNumberFamily = '',
+  } = params ?? {};
+  const policyNumberSeparator =
+    policyNumberPerson && policyNumberFamily ? SEPARATORS.POLICY_NUMBER : '';
+  const policyNumber = policyNumberPerson + policyNumberSeparator + policyNumberFamily;
+  const queryParams = [
+    { first_name: firstName },
+    { last_name: lastName },
+    { code: registrationCode },
+    { dob: dateOfBirth },
+    { policy_number: policyNumber },
+  ];
+  return queryParams.reduce((queryString, param) => {
+    const [[key, value]] = Object.entries(param);
+    if (!value) return queryString;
+    const paramString = `${key}=${value}`;
+    const paramSeparator = queryString.length > 0 ? SEPARATORS.PARAMETERS : SEPARATORS.QUERY_STRING;
+    return queryString + paramSeparator + paramString;
+  }, '');
+};
+
+const getPatientRequestUrl = params => {
+  const baseUrl = UIDatabase.getSetting(SYNC_URL);
+  const endpoint = RESOURCES.PATIENT;
+  const queryString = getQueryString(params);
+  return baseUrl + endpoint + queryString;
+};
+
+const getPrescriberRequestUrl = params => {
+  const baseUrl = UIDatabase.getSetting(SYNC_URL);
+  const endpoint = RESOURCES.PRESCRIBER;
+  const queryString = getQueryString(params);
+  return baseUrl + endpoint + queryString;
+};
+
+const processPatientResponse = async response => {
+  const responseJson = await response.json();
+  const { error } = responseJson;
+  if (error) throw new Error(error);
+  const patientData = responseJson.map(patient => ({
```

Comment:
Not sure how the api is working on desktop - is this the full record? so you just need to translate the remaining fields to have a saveable-record?

Reply:
Yeah, currently quite basic, just returns the (mostly) full record. At the moment it's basically similar to the data sync receives, so I'm hoping a lot of creation logic can be reused.